### PR TITLE
[Improve][Issue-2681] support team dependency management

### LINKED ISSUE #2681: [Feature] support dependency management
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

StreamaPark supports users upload jar when editing application, It would more convenient if we provide a easy way to manage dependency. For example, the users in the same team can share all the dependencies, because the dependencies are reused in team, maybe we can save a lot disk space.

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhoulii**
## What changes were proposed in this pull request

Issue Number: close #2681 

## Verifying this change

- *Manually verified the change by testing locally.* 

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): (yes / **no**)

### CODE REVIEW

**Comment by GOODBOY008** [comment]
I think `dependency management` should have a refer list which job use this .When modify can change `job` status to notify user resubmit and make change effitive , just like `varible management`. When delete a dependency which was used can alert `delete can not complete with job refer`.

**Comment by zhoulii** [comment]
> I think `dependency management` should have a refer list which job use this .When modify can change `job` status to notify user resubmit and make change effitive , just like `varible management`. When delete a dependency which was used can alert `delete can not complete with job refer`.

Nice suggestions, will add this feature.

**Comment by zhoulii** [comment]
This PR is ready, Please take a look, thanks. @wolfboys

**Comment by wolfboys** [comment]
> This PR is ready, Please take a look, thanks. @wolfboys

Thanks for your contribution, Great feature, I will review it later.

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql [hunk 1]
```diff
@@ -31,6 +31,24 @@ CREATE TABLE `t_external_link` (
   `modify_time` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP COMMENT 'modify time'
 ) engine = innodb default charset=utf8mb4 collate=utf8mb4_general_ci;
 
+-- ----------------------------
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: The upgraded SQL should be in 2.2.0.sql

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql [hunk 2]
```diff
@@ -29,6 +29,40 @@ create table "public"."t_external_link" (
 "modify_time" timestamp(6) not null default timezone('UTC-8'::text, (now())::timestamp(0) without time zone)
 );
 
+-- ----------------------------
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: The upgraded SQL should be in 2.2.0.sql

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql (hunk 1 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/assembly/script/upgrade/mysql/2.1.0.sql]: done.

File: streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql (hunk 2 above)
Comment (zhoulii) [reply streampark-console/streampark-console-service/src/main/assembly/script/upgrade/pgsql/2.1.0.sql]: done.

**Review by wolfboys — APPROVED** [review]
Thanks for your contributions. Overall, looks good. We can merge them first, and address any more detailed issues later.

**Pull request merged**